### ISSUE TITLE
Not be able to assign value dynamically for the match_array function .

### ISSUE THREAD

**Issue by RVaidhy** [openobserve/openobserve#1448]
### Which OpenObserve functionalities are the source of the bug?

ingestion

### Is this a regression?

Yes

### Description

[match_array]

Match at least one element
Source
	match_array(["foobar", "bazqux"], r'foo')
Return
true

How to assign values instead of foo dynamically from a variable. Please help. 

### Please provide a link to a minimal reproduction of the bug

https://playground.vrl.dev/?state=eyJwcm9ncmFtIjoiXG5cbi5lcnJvcnMgPSBbXCIuZ2l0XCIsXCJpbmZvXCIsXCJvc3BsYXRmb3JtXCJdXG5cbi5zYW1wbGVyZXN1bHQgPSBtYXRjaF9hcnJheSguZXJyb3JzLHInc3Nzc3NzJylcblxuLnJlcV91cmxfc3BsaXQsZXJyb3IxPXNwbGl0KC5yZXF1ZXN0X1VSTCwgXCIvXCIpXG5cbi5tYWluID0gLnJlcV91cmxfc3BsaXRbMV1cblxuLnJlc3VsdHNzc3MgPSBtYXRjaF9hcnJheSguZXJyb3JzLHInLmdpdCcpXG5cblxuIyBOb3Qgd29ya2luZyBzY2VuYXJpb1xuIyBwbGVhc2UgdW5jb21tZW50IGFuZCBjaGVjayB0aGUgcmVzdWx0c1xuIyBJIHdhbnQgdG8gdmFsaWRhdGUgZm9yIC5tYWluIHZhbHVlIGR5bmFtaWNhbGx5IGV2ZXJ5dGltZVxuXG4jLnJlc3VsdHNzc3MgPSBtYXRjaF9hcnJheSguZXJyb3JzLC5tYWluKVxuXG4iLCJldmVudCI6eyJ0aW1lc3RhbXAiOiIyMDIzLTA4LTIzVDEwOjQ5OjU5LjQwOVRaIiwiaWQiOiIzMnFxamhhc2QzNHEzNG4iLCJ1c2VybmFtZSI6ImFub255bW91cyIsInJlcXVlc3RfbWV0aG9kIjoiR0VUIiwicmVxdWVzdF9VUkwiOiIvLmdpdC9pbmZvLyIsImFnZW50IjoiSmF2YSIsImh0dHBfc3RhdHVzIjoiMjIwIn0sImlzX2pzb25sIjpmYWxzZSwiZXJyb3IiOm51bGx9

### Please provide the exception or error you saw

```true
Hello Team,


I am trying to parse logs using the function.
I have a scenario where I need to identify the url's which starts with specific words and needs to mark that as system calls. 

So planning to use match_array. Where I will key in the specific words as part of the array and when every row comes I shall extract the word and compare. Here if I give the input inside r'' ( in VRL playgroud I am testing ) it is working. but if i try to assign the value it is not working . Please help me to resolve this. 

https://vector.dev/docs/reference/vrl/functions/#match_array

Here is the link to see for sample input and expected output :- 

https://playground.vrl.dev/?state=eyJwcm9ncmFtIjoiXG5cbi5lcnJvcnMgPSBbXCIuZ2l0XCIsXCJpbmZvXCIsXCJvc3BsYXRmb3JtXCJdXG5cbi5zYW1wbGVyZXN1bHQgPSBtYXRjaF9hcnJheSguZXJyb3JzLHInc3Nzc3NzJylcblxuLnJlcV91cmxfc3BsaXQsZXJyb3IxPXNwbGl0KC5yZXF1ZXN0X1VSTCwgXCIvXCIpXG5cbi5tYWluID0gLnJlcV91cmxfc3BsaXRbMV1cblxuLnJlc3VsdHNzc3MgPSBtYXRjaF9hcnJheSguZXJyb3JzLHInLmdpdCcpXG5cblxuIyBOb3Qgd29ya2luZyBzY2VuYXJpb1xuIyBwbGVhc2UgdW5jb21tZW50IGFuZCBjaGVjayB0aGUgcmVzdWx0c1xuIyBJIHdhbnQgdG8gdmFsaWRhdGUgZm9yIC5tYWluIHZhbHVlIGR5bmFtaWNhbGx5IGV2ZXJ5dGltZVxuXG4jLnJlc3VsdHNzc3MgPSBtYXRjaF9hcnJheSguZXJyb3JzLC5tYWluKVxuXG4iLCJldmVudCI6eyJ0aW1lc3RhbXAiOiIyMDIzLTA4LTIzVDEwOjQ5OjU5LjQwOVRaIiwiaWQiOiIzMnFxamhhc2QzNHEzNG4iLCJ1c2VybmFtZSI6ImFub255bW91cyIsInJlcXVlc3RfbWV0aG9kIjoiR0VUIiwicmVxdWVzdF9VUkwiOiIvLmdpdC9pbmZvLyIsImFnZW50IjoiSmF2YSIsImh0dHBfc3RhdHVzIjoiMjIwIn0sImlzX2pzb25sIjpmYWxzZSwiZXJyb3IiOm51bGx9
```

**Comment by RVaidhy** [comment]
For your clarification, We are using openobserve-v0.5.1-linux-amd64-musl.tar. Please help !

**Comment by oasisk** [comment]
@RVaidhy please upgrade to v0.5.2 , to use functions with more lines.

Also please note , when you ingest data to openobserve all fields names are in small case , for eg :

{"_timestamp":1694143686586724,"agent":"Java","http_status":"220","id":"32qqjhasd34q34n","operation":"USER CALL","request_method":"GET","request_url":"/api/cargo-local/GYQV_MATANO_SAMPLE_REPO/sample.json","request_user_agent":"jfrog-mission-control/erere","timestamp":"2023-08-23T10:49:59.409TZ","username":"anonymous"}

Hence your function needs to be modified to use:
.req_url_split,error1=split(.request_**url**, "/")

instead of : .req_url_split,error1=split(.request_**URL**, "/") to

**Comment by oasisk** [comment]
tested with above mentioned data & function, working fine with fixes

**Comment by RVaidhy** [comment]
Hi Oasisk,

Even after applying the fixes that you have mentioned we had that same issue. 

To get rid of that we have used the following method. Basically match_array expects regex formatted data. So from a full string if you extract a sub-string and wanted to validate that specific value against array or something we may need to follow the below options.

Say you have a predefined list of values and you wnated to compare the extracted value with that. This is our scenario. 
request_url_array = ["sample","values","tobe","inserted","here"]

Now, from logs we have got one field like below. 
.extrated_substring = string!(.url)

Convert that to regex as below,
.urlInfoRegex = to_regex(.extrated_substring)?? r''

Now pass this to match_array. 
.isUrlExist = match_array(.request_url_array,.urlInfoRegex)

So in our case, if we want to validate a specific value against an array of values we shall refer the above method. It may be useful to someone. Thats why sharing here for ref. Thanks!
